React/Modules/RCTAsyncLocalStorage.m

```diff
@@ -445,4 +392,31 @@ - (NSDictionary *)_writeEntry:(NSArray<NSString *> *)entry changedManifest:(BOOL
   }
 }
 
+RCT_EXPORT_METHOD(setStorageDirectory:(NSString*)storageDirectory
+                  callback:(RCTResponseSenderBlock)callback)
+{
+  storageDirectory = [storageDirectory stringByAppendingPathComponent:RCTStorageDirectory];
```

Comment:
It would probably be better to use an NSURL for the storageDirectoryParam, because it includes all the logic for expanding ~/ etc. If not, you should probably at least call stringByExpandingTildeInPath, otherwise there's no good way to specify paths in the docs or library directory.